src/R/Components/Impl/ConnectionManager/Implementation/Connection.cs

```diff
@@ -20,5 +30,43 @@ internal class Connection : ConnectionInfo, IConnection {
         /// If true, the connection is to a remote machine
         /// </summary>
         public bool IsRemote { get; }
+
+        public override string UserProvidedPath {
+            get { return base.UserProvidedPath; }
+            set {
+                base.UserProvidedPath = value;
+                Path = ToCompletePath(value);
+            }
+        }
+
+        public static string ToCompletePath(string path) {
```

Comment:
I personally don't want to see 'auto-correction'. If I entered 'mymachine' this is what I want to see in UI. Internal details should be left to the code to handle. I don't want edit box to suddenly change to full URL when I click Save or restart VS.